overlord/snapstate/handlers_components.go

```diff
@@ -266,18 +266,19 @@ func (m *SnapManager) doLinkComponent(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	cs := sequence.NewComponentState(compSetup.CompSideInfo, compSetup.CompType)
 	// set information for undoLinkComponent in the task
-	t.Set("linked-component", csi)
+	t.Set("linked-component", cs)
 	// Append new component to components of the current snap
-	if err := snapSt.Sequence.AddComponentForRevision(snapInfo.Revision, csi); err != nil {
+	if err := snapSt.Sequence.AddComponentForRevision(snapInfo.Revision, cs); err != nil {
 		return fmt.Errorf("internal error while linking component: %w", err)
 	}
 
 	// Store last component refresh time
 	if snapSt.LastCompRefreshTime == nil {
 		snapSt.LastCompRefreshTime = make(map[string]time.Time)
 	}
-	snapSt.LastCompRefreshTime[csi.Component.ComponentName] = timeNow()
+	snapSt.LastCompRefreshTime[compSetup.CompSideInfo.Component.ComponentName] = timeNow()
```

Comment:
seems maybe ComponentSetup should grow a ComponentName method?

Reply:
Added